src/components/area-plot/components/stacked-data-annotation/StackedDataAnnotation.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { dataChangeTypes } from '../../../../constants/plotConstants';
+import DataAnnotation from '../../../shared/data-annotation/DataAnnotation';
+
+function getFormattedPercentageValue({ currentValue, getTotalValue, index }) {
+  const totalValue = getTotalValue(index);
+
+  if (totalValue === 0) {
+    return '0%';
+  }
+
+  return `${((100 * currentValue) / totalValue).toFixed(1)}%`;
+}
+
+export default function StackedDataAnnotation({
+  x,
+  y,
+  stroke,
+  valueFormatter,
+  dataChangeType,
+  index,
+  getTotalValue,
+  // The `value` prop that Recharts passes this component is the "cumulative" value,
+  // so if it comes from a line that is stacked on others, it will be bigger than the
+  // actual value we want to display in the annotation, so we use this `getCurrentValue` prop.
+  getCurrentValue,
+  dataKey,
+}) {
+  const currentValue = getCurrentValue(index, dataKey);
+
+  const changeTypeToFormattingFunc = {
```

Comment:
I'm trying to avoid ternaries when it's less of a boolean decision and more of a "is it Red or Blue?"-type decision and instead use these object mappings. The syntax still looks strange but IMO it's clearer than other control flow approaches.